### PR TITLE
dns: improved handling of corrupt additionals

### PR DESCRIPTION
**Pull request by catenacyber**
Link to ticket: https://redmine.openinfosecfoundation.org/issues/
https://redmine.openinfosecfoundation.org/issues/7228

Describe changes:
- dns: improved handling of corrupt additionals

### Provide values to any of the below to override the defaults.

SV_BRANCH=https://github.com/OISF/suricata-verify/pull/2032

### CODE REVIEW

**Comment by victorjulien** [comment]
Not sure I fully understand. Is this about logging a partial corrupt message?

**Comment by catenacyber** [comment]
> Is this about logging a partial corrupt message?

yes

File: rules/dns-events.rules [hunk 1]
```diff
@@ -8,3 +8,4 @@ alert dns any any -> any any (msg:"SURICATA DNS Not a response"; flow:to_client;
 # Z flag (reserved) not 0
 alert dns any any -> any any (msg:"SURICATA DNS Z flag set"; app-layer-event:dns.z_flag_set; classtype:protocol-command-decode; sid:2240006; rev:2;)
 alert dns any any -> any any (msg:"SURICATA DNS Invalid opcode"; app-layer-event:dns.invalid_opcode; classtype:protocol-command-decode; sid:2240007; rev:1;)
+alert dns any any -> any any (msg:"SURICATA DNS corrupt additionals"; app-layer-event:dns.corrupt_additionals; classtype:protocol-command-decode; sid:2240008; rev:1;)
```
Comment (jasonish) [inline rules/dns-events.rules:11]: Do we use `invalid` usually?

Comment (catenacyber) [reply rules/dns-events.rules:11]: I will update

File: rust/src/dns/parser.rs [hunk 2]
```diff
@@ -362,7 +362,11 @@ pub fn dns_parse_body<'a>(
     let (i, queries) = count(|b| dns_parse_query(b, message), header.questions as usize)(i)?;
     let (i, answers) = dns_parse_answer(i, message, header.answer_rr as usize)?;
     let (i, authorities) = dns_parse_answer(i, message, header.authority_rr as usize)?;
-    let (i, additionals) = dns_parse_answer(i, message, header.additional_rr as usize)?;
+    let additionals_parsed = dns_parse_answer(i, message, header.additional_rr as usize);
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply rust/src/dns/parser.rs:365]: It looks a good idea. I began with the last one

**Comment by suricata-qa** [comment]
ERROR:
 

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
  

Pipeline 22502

**Comment by catenacyber** [comment]
Continued in #11752

**Pull request closed without merging**